lib/atomic_time/core_ext/active_record/type.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'active_record/type'
+
+module ActiveRecord
+  module Type
+    class AtomicTime < Value
+      def type
+        :atomic_time
+      end
+
+      def deserialize(value)
+        return value if value.nil?
+
+        ::AtomicTime.at value
+      end
+
+      def cast(value)
+        value&.to_atomic_time
+      end
+
+      def serialize(value)
+        value&.to_i
+      end
+    end
+
+    register :atomic_time, ::ActiveRecord::Type::AtomicTime
```

Comment:
Yeah - this line is the reason I stuck it here. I thought about separating this line from the class definition which is still valid in my mind - I think I ended up with just putting them both here as you could construe a new "type" as an "extension" to active record.

Reply:
Yeah, I think (as much as possible) classes should be found in a file with a matching name.

If you follow my advice about including modules rather than reopening classes, then this can _also_ go in a `ActiveSupport.on_load(:active_record)` block somewhere.